students/AndrewMiotke/lesson03/assignment/src/basic_operations.py

```diff
@@ -0,0 +1,126 @@
+""" database assignment """
+
+import logging
+import peewee
+
+database = peewee.SqliteDatabase('customers.db')
+# database.execute_sql('PRAGMA foreign_keys=ON;')
+database.connect()
+database.execute_sql('drop table if exists customer;')
+
+
+# Logging just for fun
+LOG_FILE = "database_logs.log"
+LOGGER = logging.getLogger()
+LOGGER.setLevel(logging.ERROR)
+LOG_FORMAT = "%(asctime)s\t %(filename)s: %(lineno)-3d %(levelname)s %(message)s)"
+FORMATTER = logging.Formatter(LOG_FORMAT)
+
+FILE_HANDLER = logging.FileHandler(LOG_FILE)
+FILE_HANDLER.setFormatter(FORMATTER)
+
+class BaseModel(peewee.Model):
+    """ Base class for peewee and the sqlite3 database """
+
+    class Meta:
+        """ Peewee boilerblate """
+        database = database
+
+
+class Customer(BaseModel):
+    """ Establish the datbase model """
+
+    customer_id = peewee.CharField(primary_key=True, max_length=30)
+    name = peewee.CharField(max_length=50)
+    lastname = peewee.CharField(max_length=50)
+    address = peewee.CharField(max_length=75)
+    phone_number = peewee.CharField(max_length=15)
+    email = peewee.CharField(max_length=320)
+    status = peewee.CharField(max_length=10)
+    credit_limit = peewee.FloatField()
+
+
+database.create_tables([Customer])
+
+
+def add_customer(customer_id, name, lastname, address, phone_number, email, status, credit_limit):
+    """ Add a customer to the database """
+
+    try:
+        with database.transaction():
+            Customer.create(
+                customer_id=customer_id,
+                name=name,
+                lastname=lastname,
+                address=address,
+                phone_number=phone_number,
+                email=email,
+                status=status,
+                credit_limit=credit_limit,
+                )
+            customer.save()
+            logging.error(f"Successfully saved {customer_id} to database")
+
+    except:
+        logging.error(f"Error creating {customer_id}")
+    finally:
```

Comment:
You will want to remove all these "finally" blocks in your code. You do not want to close the database because once you close the connection, it cannot reach the database again to make changes!